src/components/network-map-tab.js

```diff
@@ -581,8 +594,13 @@ export const NetworkMapTab = ({
     ]);
 
     const updateMapEquipmentsAndGeoData = useCallback(() => {
-        updateMapEquipments().then(loadGeoData);
-    }, [updateMapEquipments, loadGeoData]);
+        const previousCurrentNode = currentNode;
```

Comment:
to avoid adding a new usecallback dependency (which is a dependency in updateMapEquipments anyway) you could use `currentNodeRef.current` instead of `currentNode`
in
```
const previousCurrentNode = currentNodeRef.current;
```
then it is also miss-named, maybe something like `currentNodeAtRelaodCalling` is a better name
To mark that we check at the end of an async call (fetch) that the context (node) is still the same.

Reply:
Done